simc.py

```diff
@@ -0,0 +1,175 @@
+    r'git build (?P<expansion>\S+) (?P<git_commit>[a-f0-9]{1,7}))?')
+_PACKAGE_RE = re.compile(
+    r'(?P<filename>simc-(?P<version>\d+-\d+)-'
+    r'(?P<platform>[^-]+)-(?P<git_commit>[a-f0-9]{1,7})\.(?:dmg|7z))')
+
+
+@dataclass(frozen=True)
+class SimcVersion:
+    version: str
+    wow_version: str
+    hotfix_date: datetime.date
+    hotfix_version: int
+    expansion: Optional[str] = None
+    git_commit: Optional[str] = None
+
+    @staticmethod
+    def parse(input: str) -> 'Optional[SimcVersion]':
+        match = _VERSION_RE.search(input)
+        if not match:
+            return
+        
+        match = match.groupdict()
+        match['hotfix_version'] = int(match['hotfix_version'])
+        match['hotfix_date'] = datetime.date.fromisoformat(match['hotfix_date'])
+        return SimcVersion(**match)
+
+
+class LinkParser(html.parser.HTMLParser):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.links = []  # type: List[str]
+
+    def handle_starttag(self, tag, attrs):
+        if tag != 'a':
+            return
+        attrs = dict(attrs)
+        href = attrs.get('href')
+        if href and '/' not in href:
+            self.links.append(href)
+
+
+def get_simc_version(simc_path: str) -> Optional[SimcVersion]:
+    """Gets simc version information."""
+    p = subprocess.run([simc_path], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    return SimcVersion.parse(p.stdout.decode())
+
+
+def simc_platform() -> Optional[str]:
+    """SimC platform name for binary builds."""
+    if platform.system() == 'Windows':
+        return {
+            'AMD64': 'win64',
+            'x86': 'win32',
+        }.get(platform.machine())
+    elif platform.system() == 'Darwin':
+        return 'macos'
+
+
+def latest_simc_version(major_ver: str = _CURRENT_SIMC_VERSION, platform: Optional[str] = None) -> Optional[Tuple[str, str]]:
+    """
+    Checks the SimulationCraft nightly builds for the latest binary build.
+    
+    Args:
+        major_ver: A major version of simc, eg: '910-01'.
+        platform: Platform to check, eg: 'win64', 'macos', or None to auto-detect.
+
+    Returns:
+        tuple of (filename, git_commit) with the latest version.
+    """
+    if not platform:
+        platform = simc_platform()
+        if not platform:
+            # Unsupported platform, cannot answer.
+            return
+
+    listing = urlopen('http://downloads.simulationcraft.org/nightly/?C=M;O=D').read().decode('utf-8')
+    parser = LinkParser()
+    parser.feed(listing)
+
+    # Parse the filenames
+    for link in parser.links:
+        v = _PACKAGE_RE.match(link)
+        if not v:
+            continue
+        if major_ver == v.group('version') and platform == v.group('platform'):
+            return (v.group('filename'), v.group('git_commit'))
+
+
+def download_simc():
+    if platform.system() != "Windows" or not platform.machine().endswith('64'):
```

Comment:
I would remove that check, to remove looking at the system platform multiple times and artificially restricting the selection.
latest_simc_verison() already access simc_platform() and if that returns None, the platform is not supported. Maybe move the print message to that.